packages/functions/src/donor/CompleteAppointmentHandler.ts

```diff
@@ -68,6 +70,11 @@ export async function completeAppointmentFunc(
 
   await AppointmentDataAccessLayer.setAppointment(updatedAppointment);
 
+  if (appointment.status == AppointmentStatus.COMPLETED) {
+    // TODO : verify if donorId is valid, should be done here or in DAL ?
```

Comment:
I think you should validate here.

BTW, if donorId is MANUAL_DONOR_ID I think you should not run this update